addons/mail/static/src/components/composer/composer.scss

```diff
@@ -174,19 +184,11 @@
     &.o-has-current-partner-avatar {
 
         .o_Composer_actionButtons.o-composer-is-compact > *:last-child {
-            border-radius: 0 10px 10px 0;
+            border-radius: 0 $o-mail-rounded-rectangle-border-radius-lg $o-mail-rounded-rectangle-border-radius-lg 0;
         }
 
-        .o_Composer_textInput {
-            border-radius: 10px 0 0 10px;
-
-            > .note-editor {
-                border-radius: 10px 0 0 10px;
-
-                > .note-editing-area {
-                    border-radius: 10px 0 0 10px;
-                }
-            }
+        .o_Composer_textInput.o-composer-is-compact {
+            border-radius: $o-mail-rounded-rectangle-border-radius-lg 0 0 $o-mail-rounded-rectangle-border-radius-lg;
```

Comment:
Would flatten style rules by having `o-has-current-partner-avatar` modifier on `.o_Composer_actionButton-last` & `.o_Composer_textInput`. And replace `> *:last-child` by a `.o_Composer_actionButton-last` (or `o-last`, but likely would accidentally share style) modifier.